ast/dml.go

```diff
@@ -476,6 +476,8 @@ type SelectStmt struct {
 	LockTp SelectLockType
 	// TableHints represents the table level Optimizer Hint for join type
 	TableHints []*TableOptimizerHint
+	// UnionDistinct is mark flag that previous union is distinct
+	UnionDistinct bool
```

Comment:
s/UnionDistinct/HasUnionDistinct/?

Reply:
Finally, I use `IsAfterUnionDistinct` to indicates whether it's a stmt after "union distinct" one